tests/test_integration/stencil_definitions.py

```diff
@@ -374,3 +384,29 @@ def two_optional_fields(
             out_a = out_a + dt * phys_tend_a
         if __INLINED(PHYS_TEND_B):
             out_b = out_b + dt * phys_tend_b
+
+
+@register
+def produce_higher_dimensional_field(input_field: Field3D, output_field: Field4D):
+    with computation(PARALLEL), interval(...):
+        output_field[0, 0, 0][0] = input_field
+
+
+@register
+def consume_higher_dimensional_field(input_field: Field3D, output_field: Field4D):
+    with computation(PARALLEL), interval(...):
+        input_field = output_field[0, 0, 0][0]
+
+
+@register
+def produce_higher_dimensional_field_indirect_addressing(
+    input_field: Field3D, output_field: Field4D, variable: float
```

Comment:
`float` is not the right type for indirect accesses and should throw an error. Please use an unsigned type and add a test checking if non unsigned types throw a proper exception.

Reply:
Fixed. Good catch!!